201-timeseriesinsights-environment-with-longterm-storage/azuredeploy.json

```diff
@@ -0,0 +1,294 @@
+            "name": "[parameters('eventSourceName')]",
+            "apiVersion": "2018-08-15-preview",
+            "location": "[parameters('location')]",
+            "kind": "Microsoft.IoTHub",
+            "properties": {
+              "eventSourceResourceId": "[variables('eventSourceResourceId')]",
+              "iotHubName": "[parameters('iotHubName')]",
+              "consumerGroupName": "[parameters('consumerGroupName')]",
+              "keyName": "[parameters('eventSourceKeyName')]",
+              "sharedAccessKey": "[listkeys(concat(variables('eventSourceResourceId'), '/IoTHubKeys/', parameters('eventSourceKeyName')), '2016-02-03').primaryKey]",
+              "timestampPropertyName": "[parameters('eventSourceTimestampPropertyName')]"
+            },
+            "tags": "[variables('eventSourceTagsValue')]",
+            "dependsOn": [
+              "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]",
+              "[concat('Microsoft.Devices/IotHubs/', parameters('iotHubName'))]",
+              "[concat('Microsoft.Devices/IotHubs/', parameters('iotHubName'), '/eventHubEndpoints/events/ConsumerGroups/', parameters('consumerGroupName'))]"
+            ]
+          }
+        ]
+      },
+      {
+        "condition": "[not(empty(parameters('accessPolicyReaderObjectIds')))]",
+        "type": "Microsoft.TimeSeriesInsights/environments/accesspolicies",
+        "name": "[concat(parameters('environmentName'), '/', 'readerAccessPolicy', copyIndex())]",
+        "copy": {
+          "name": "accessPolicyReaderCopy",
+          "count": "[if(empty(parameters('accessPolicyReaderObjectIds')), 1, length(parameters('accessPolicyReaderObjectIds')))]"
+        },
+        "apiVersion": "2018-08-15-preview",
+        "properties": {
+          "principalObjectId": "[parameters('accessPolicyReaderObjectIds')[copyIndex()]]",
+          "roles": [
+            "Reader"
+          ]
+        },
+        "dependsOn": [
+          "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]"
+        ]
+      },
+      {
+        "condition": "[not(empty(parameters('accessPolicyContributorObjectIds')))]",
+        "type": "Microsoft.TimeSeriesInsights/environments/accesspolicies",
+        "name": "[concat(parameters('environmentName'), '/', 'contributorAccessPolicy', copyIndex())]",
+        "copy": {
+          "name": "accessPolicyReaderCopy",
+          "count": "[if(empty(parameters('accessPolicyContributorObjectIds')), 1, length(parameters('accessPolicyContributorObjectIds')))]"
+        },
+        "apiVersion": "2018-08-15-preview",
+        "properties": {
+          "principalObjectId": "[parameters('accessPolicyContributorObjectIds')[copyIndex()]]",
+          "roles": [
+            "Contributor"
+          ]
+        },
+        "dependsOn": [
+          "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]"
+        ]
+      },
+      {

```

Comment:
actually if this is the only difference in templates (and the few property references above) - then let's combine these into a single sample and remove the duplication...

Reply:
@bmoore-msft I agree that there is some duplication here. One way to clean this up would be to reduce these three quickstarts to two: 1) S Sku + Event Hub and 2) L Sku +IoT Hub

The TSI environment properties have changed since introduction L Sku's and creating conditional logic to populate each would introduce some configuration overhead that most customers will never need. Also, L Sku + IoT Hub is the configuration we suggest moving forward as we roll out L Sku's for GA. The S Sku + Event Hub would remain as a guide post for users that would like to still configuration TSI that route.